packages/cli/src/commands/stories/push/index.ts

```diff
@@ -0,0 +1,288 @@
+            processProgress.setTotal(total);
+            updateProgress.setTotal(total);
+          },
+          onStoryError(error) {
+            summary.creationResults.failed += 1;
+            summary.processResults.total -= 1;
+            summary.updateResults.total -= 1;
+            processProgress.setTotal(summary.processResults.total);
+            updateProgress.setTotal(summary.updateResults.total);
+            ui.error('Failed to process local story', error);
+            logger.error('Failed to process local story', { error });
+          },
+        }),
+        // Map all references to numeric ids and uuids.
+        mapReferencesStream({
+          schemas,
+          maps,
+          onIncrement() {
+            processProgress.increment();
+          },
+          onStorySuccess(localStory, processedFields, missingSchemas) {
+            warnAboutCustomPlugins(processedFields, localStory);
+            warnAboutMissingSchemas(missingSchemas, localStory);
+            logger.info('Processed story', { storyId: localStory.uuid });
+            summary.processResults.succeeded += 1;
+          },
+          onStoryError(error, localStory) {
+            summary.processResults.failed += 1;
+            summary.updateResults.total -= 1;
+            const message = 'Failed to map story references';
+            ui.error(message, error);
+            logger.error(message, { error, storyId: localStory.uuid });
+            updateProgress.setTotal(summary.updateResults.total);
+          },
+        }),
+        // Update remote stories with correct references.
+        writeStoryStream({
+          transport: options.dryRun
+            ? { write: (story: Story) => Promise.resolve(story) }
+            : makeWriteStoryAPITransport({
+                spaceId: space,
+                publish: options.publish,
+              }),
+          onIncrement() {
+            updateProgress.increment();
+          },
+          onStorySuccess(localStory) {
+            logger.info('Updated story', { storyId: localStory.uuid });
+            summary.updateResults.succeeded += 1;
+          },
+          onStoryError(error, localStory) {
+            summary.updateResults.failed += 1;
+            const message = 'Failed to update story';
+            ui.error(message, error);
+            logger.error(message, { error, storyId: localStory.uuid });
+          },
+        }),
+      );
+    }
+    catch (maybeError) {
+      handleError(toError(maybeError));
+    }
+    finally {
+      logger.info('Pushing stories finished', summary);
+      ui.stopAllProgressBars();
+      const failedStories = Math.max(summary.creationResults.failed, summary.processResults.failed, summary.updateResults.failed);
+      ui.info(`Push results: ${summary.creationResults.total} ${summary.creationResults.total === 1 ? 'story' : 'stories'} pushed, ${failedStories} ${failedStories === 1 ? 'story' : 'stories'} failed`);
```

Comment:
nitpick: `creationResults` are the stories placeholders created, but not the full story pushed, correct? If so, maybe we can instead inform with something like `stories succeded/skipped` to avoid confusion?